src/viewport/viewport.js

```diff
@@ -6,26 +6,39 @@
 // mapbox-gl's implementation to ensure that seamless interoperation
 // with mapbox and react-map-gl.
 // See: transform.js in https://github.com/mapbox/mapbox-gl-js
-
-import Viewport, {COORDINATE_SYSTEM}
-  from 'viewport-mercator-project/perspective';
+import {COORDINATE_SYSTEM} from 'viewport-mercator-project';
 import autobind from 'autobind-decorator';
 import {mat4, vec4} from 'gl-matrix';
-
-export {COORDINATE_SYSTEM}
-  from 'viewport-mercator-project/perspective';
+import assert from 'assert';
 
 function fp64ify(a) {
   const hiPart = Math.fround(a);
   const loPart = a - Math.fround(a);
   return [hiPart, loPart];
 }
 
-export default class WebGLViewport extends Viewport {
+export default class WebGLViewport {
+
+  constructor({viewport}) {
+    this.viewport = viewport;
```

Comment:
LayerManager's context.viewport property points to a WebGLViewport, which is now a wrapper around "real" viewport of type WebMercatorViewport. Some `context.viewport` references will be broken. e.g. reflection-effect.js, line 42